### PR TITLE
Fail fast

### PR DESCRIPTION
**Pull request by zhengchang907**
## Main Change
Add fail fast scripts to check:
* UAMI is assigned as 'Contributor' or 'Owner' role of the deployment subscription
* There is enough computing resource for the deployment
* If users enabled Satellite server feature, validate the Satellite can be resolved based on provided FQDN

## Test
* Deploy the preview offer (with Domain mode and without Satellite)
* [Pipeline](https://github.com/zhengchang907/rhel-jboss-templates/actions/runs/2688798026)

### CODE REVIEW

File: eap74-rhel8-payg-multivm/src/main/scripts/validate-parameters.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+function echo_stderr() {
+    echo "$@" 1>&2
+    # The function is used for scripts running within Azure Deployment Script
```
Comment (edburns) [inline eap74-rhel8-payg-multivm/src/main/scripts/validate-parameters.sh:5]: Why no

```
#!/usr/bin/env bash
set -Eeuo pipefail
```

**Review by edburns — APPROVED** [review]
See comments inline.

**Review by edburns — CHANGES_REQUESTED** [review]
See comments inline.

**Comment by zhengchang907** [comment]
Hi @edburns, I added the flag in the latest [commit](https://github.com/Azure/rhel-jboss-templates/pull/92/commits/a558195f50294c106c7cd1debeef87568c65bf0d)

**Review by edburns — APPROVED** [review]

**Pull request merged**